great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -422,7 +422,7 @@ def batch_parameters_to_batch_spec_kwarg_identifiers(
 ]
 
 
-class _SQLAsset(DataAsset):
+class _SQLAsset(DataAsset[Partitioner]):
```

Comment:
I have to imagine there's something we need to do here for the datasource type. Maybe just pass in `_DatasourceT`?